scipy/optimize/optimize.py

```diff
@@ -1088,6 +1088,40 @@ def fmin_bfgs(f, x0, fprime=None, args=(), gtol=1e-5, norm=Inf,
     allvecs  :  list
         The value of xopt at each iteration. Only returned if retall is True.
 
+    Examples
+    --------
+    >>> from scipy.optimize import fmin_bfgs
+
+    >>> def rosenbrock(x):
```

Comment:
Could you remove the inline version of rosenbrock? The use of minimize over the `fmin_*` methods is probably preferred, and so the full function is probably not needed.

Reply:
Drive-by 2c comment: a spelled-out 2D rosenbrock function would be very helpful for me as a user; way more so than an imported version. The latter does not help answering "how do I define a function to be minimized".

Reply:
Yes, this redundant definition is from my experience when I was new to use fmin_*. Looking back at the time, it would be helpful if an example includes how to use 'args'. 

If the duplicated definition of the Rosebrock function is the matter, what about replacing this part as follow?

```
def quadratic_objective(x, Q):
    return x.T@Q@x
fmin_bfgs(quadratic_objective, x0, args=(np.diag([1., 10.]),))
```

This totally depends on my personal experience, so if this is not the case for other people, I just rewrite this to a predefined Rosenbrock version.

Reply:
One last nit: if you go with a quadratic objective example, separate the array definition into a statement of its own. Otherwise the tuple syntax with a trailing comma is hard to parse (and this is a common source of issues for people)

Reply:
Updated the example. I noted the trailing comma though this might be verbose.

Reply:
Looks good to me now. (I _think_ a single #may vary would be enough.) 
I'll defer to @andyfaff for the decision